src/test/java/com/github/alexcojocaru/mojo/elasticsearch/RunElasticsearchDataMojoTest.java

```diff
@@ -0,0 +1,89 @@
+package com.github.alexcojocaru.mojo.elasticsearch;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.testing.AbstractMojoTestCase;
+import org.junit.Ignore;
+
+import java.io.File;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * @author gfernandes
+ */
+public class RunElasticsearchDataMojoTest extends AbstractMojoTestCase
+{
+
+    @Override
+    protected void setUp() throws Exception
+    {
+        super.setUp();
+    }
+    
+    @Override
+    protected void tearDown() throws Exception
+    {
+        super.tearDown();
+        
+        ElasticSearchNode.stop();
+    }
+    
+    public void testMojoLookup() throws Exception
+    {
+        File testPom = new File(getBasedir(), "src/test/resources/goals/run/pom.xml");
+
+        RunElasticSearchNodeMojo mojo = (RunElasticSearchNodeMojo)lookupMojo("run", testPom);
+ 
+        assertNotNull(mojo);
+    }
+    
+    public void testMojoExecution() throws Exception
+    {
+        File testPom = new File(getBasedir(), "src/test/resources/goals/run/pom.xml");
+
+        final RunElasticSearchNodeMojo mojo = (RunElasticSearchNodeMojo)lookupMojo("run", testPom);
+
+        final AtomicReference<Exception> internalRunThreadException = new AtomicReference<Exception>();
+
+        assertNotNull(mojo);
+        
+        Thread toRun = new Thread(new Runnable() {
+            public void run() {
+                try {
+                    mojo.execute();
```

Comment:
Let me check this because I initially wanted to have two ways to run the goal, blocking mojo execution or with a fork so maven can complete the build and ES was still running but I realized that it can't be done that so the implementation blocks the thread waiting for the killing signal. Perhaps this test is for the other implementation with fork capability.
I'll take a look to it later. Thank you.

> On Jul 16, 2015, at 8:02 PM, alexcojocaru <notifications@github.com> wrote:
> 
> In src/test/java/com/github/alexcojocaru/mojo/elasticsearch/RunElasticsearchDataMojoTest.java:
> 
> > +    }
> > +    
> > +    public void testMojoExecution() throws Exception
> > +    {
> > +        File testPom = new File(getBasedir(), "src/test/resources/goals/run/pom.xml");
> > +
> > +        final RunElasticSearchNodeMojo mojo = (RunElasticSearchNodeMojo)lookupMojo("run", testPom);
> > +
> > +        final AtomicReference<Exception> internalRunThreadException = new AtomicReference<Exception>();
> > +
> > +        assertNotNull(mojo);
> > +        
> > +        Thread toRun = new Thread(new Runnable() {
> > +            public void run() {
> > +                try {
> > +                    mojo.execute();
> Why executing it asynchronously? That forces you to have the Thread.sleep at line 68.
> 
> —
> Reply to this email directly or view it on GitHub.
>